portal-frontend/src/app/features/create-submission-dialog/create-submission-dialog.component.ts

```diff
@@ -57,40 +61,58 @@ export class CreateApplicationDialogComponent implements OnInit, AfterViewChecke
       .filter((type) => !!type.portalLabel)
       .sort((a, b) => (a.portalLabel > b.portalLabel ? 1 : -1));
     this.submissionTypes = codes.submissionTypes.sort((a, b) => (a.code > b.code ? 1 : -1));
+    this.noticeOfIntentTypes = codes.noticeOfIntentTypes.sort((a, b) => (a.portalLabel > b.portalLabel ? 1 : -1));
   }
 
   onCancel() {
     this.dialogRef.close(false);
   }
 
-  async onSubmit() {
+  async onSubmitApplication() {
     if (this.selectedAppType && this.selectedAppType.code === 'EXCL') {
-      this.currentStepIndex++;
+      this.currentStep++;
     } else {
       await this.createApplication();
     }
   }
 
+  async onSubmitNoi() {
+    if (this.selectedAppType && this.selectedAppType.code === 'EXCL') {
+      this.currentStep++;
+    } else {
+      await this.createNoi();
+    }
+  }
+
   async onSubmitInlcExcl() {
```

Comment:
Unrelated to MR, but I just noticed there was a typo here

Reply:
Done!